### ISSUE TITLE
Error with go generate using go 1.11

### ISSUE THREAD

**Issue by AdamSLevy** [goreleaser/goreleaser#781]
**Describe the bug**

When building with go 1.11 the following error occurs on the go generate step. This occurs regardless of whether the repo is built from inside or outside the GOPATH.

```
$ go generate ./...
stringer: checking package: artifact.go:5:2: could not import crypto/sha256 (type-checking package "crypto/sha256" failed (/usr/lib/go/src/crypto/sha256/sha256.go:10:2: could not import crypto (type-checking package "crypto" failed (/usr/lib/go/src/crypto/crypto.go:9:2: could not import hash (type-checking package "hash" failed (/usr/lib/go/src/hash/hash.go:8:8: could not import io (type-checking package "io" failed (/usr/lib/go/src/io/pipe.go:12:2: could not import sync (type-checking package "sync" failed (/usr/lib/go/src/sync/pool.go:9:2: could not import runtime (type-checking package "runtime" failed (/usr/lib/go/src/runtime/alg.go:8:2: could not import internal/cpu (type-checking package "internal/cpu" failed (/usr/lib/go/src/internal/cpu/cpu_x86.go:9:7: CacheLineSize redeclared in this block))))))))))))))
internal/artifact/artifact.go:15: running "stringer": exit status 1
```

**To Reproduce**

Install go 1.11. Attempt to build using `make setup build`. You will get the above output when it reaches the go generate step.

**Expected behavior**
It should build the project successfully.

**Environment (please complete the following information):**

- OS: Arch Linux
- OS version: `Linux [HOSTNAME REDACTED] 4.18.5-zen1-1-zen #1 ZEN SMP PREEMPT Fri Aug 24 12:49:14 UTC 2018 x86_64 GNU/Linux`
- GoReleaser Version [goreleaser --version] latest commit on master: 93a0055

**Comment by caarlos0** [comment]
refs #779

**Comment by caarlos0** [comment]
refs https://github.com/golang/go/issues/24661 

also check https://go-review.googlesource.com/c/tools/+/126535

**Comment by caarlos0** [comment]
should be fixed by #802
